src/promptflow/promptflow/_sdk/_start_trace.py

```diff
@@ -0,0 +1,45 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import os
+import uuid
+
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+from promptflow._sdk._constants import TRACE_SESSION_ID_ENV_VAR
+
+
+def start_trace():
+    """Start a tracing session.
+
+    This will capture OpenAI and prompt flow related calls and persist traces;
+    it will also provide a UI url for user to visualize traces details.
+
+    Note that this function is still under preview, and may change at any time.
+    """
+    # detect PFS liveness
+    pfs_port = "55507"  # TODO: make this dynamic from PFS liveness probe
+    # provision a session
+    _provision_session()
+    # init the global tracer with endpoint, context (session, run, exp)
+    _init_otel_trace_exporter(otlp_port=pfs_port)
+    # print user the UI url
```

Comment:
TODO: generate a dynamic HTML to demonstrate a list of JSON (traces)

Reply:
updated one version with mock trace UI:

![image](https://github.com/microsoft/promptflow/assets/38847871/8954b121-068c-4a5e-9ce5-0a2177612b9a)